### PR TITLE
Resizable game window support.

### PR DESCRIPTION
**Pull request by IceReaper**
### WIP!

This PR implements resizable window support. The basic functionality is tested on windows and linux (someone with a mac might verify if this works there too!).

It currently lacks the default widgets to adopt the resize method and reposition its children if necessary, so some widgets might break their bounds while resizing.
Fell free to contribute with widget implementations!

### CODE REVIEW

**Comment by pchote** [comment]
It would be good if we could get some people to test this and note all the places that break in the default mods.  This doesn't require coding knowledge, just the ability to compile and run this PR.

File: OpenRA.Game/Widgets/Widget.cs [hunk 1]
```diff
@@ -242,26 +245,16 @@ public virtual void Initialize(WidgetArgs args)
...
 
-			var substitutions = args.ContainsKey("substitutions") ?
+			substitutions = args.ContainsKey("substitutions") ?
 				new Dictionary<string, int>((Dictionary<string, int>)args["substitutions"]) :
 				new Dictionary<string, int>();
 
-			substitutions.Add("WINDOW_RIGHT", Game.Renderer.Resolution.Width);
-			substitutions.Add("WINDOW_BOTTOM", Game.Renderer.Resolution.Height);
 			substitutions.Add("PARENT_RIGHT", parentBounds.Width);
 			substitutions.Add("PARENT_LEFT", parentBounds.Left);
```
Comment (pchote) [inline OpenRA.Game/Widgets/Widget.cs]: The parent position and bounds may be defined in terms of the window size, so aren't these also going to change?

**Review by pchote — CHANGES_REQUESTED** [review]
A couple of specific comments, and one general one: i'm pretty sure that `ChromeLogic` objects will need to know when the window changes to be able to adjust bounds that it managed directly.  Can you please add plumbing for a `WindowSizeChanged` notification (there are already calls for `BecameHidden` / `BecameVisible`).

**Comment by SoScared** [comment]
This is great. How about support for adjustable widgets too? 😃

**Comment by pchote** [comment]
This was found to break rendering on HiDPI displays and had to be reverted from the KKnD mod.

Theres not much more we can do here until the logic is rewritten, so closing.

**Pull request closed without merging**